### PR TITLE
feat(python): add clients CTS

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-1729

### Changes included:

add clients CTS for Python and fixes the user agent

I will do some cleanup of the transporter in a later pr

### CODE REVIEW

**Comment by shortcuts** [comment]
Somehow scala user agent is all lowercase 🤔 is this expected @aallam or should I go ahead and capitalize like every other languages?

File: generators/src/main/java/com/algolia/codegen/cts/lambda/CodeSnakeCaseLambda.java
Comment (shortcuts) [inline generators/src/main/java/com/algolia/codegen/cts/lambda/CodeSnakeCaseLambda.java:1]: I think you'll need that too @millotp

File: tests/CTS/methods/requests/search/deleteRule.json [hunk 1]
```diff
@@ -8,5 +8,15 @@
       "path": "/1/indexes/indexName/rules/id1",
       "method": "DELETE"
     }
+  },
+  {
```
[the remark this answers is not in the document]
Comment (shortcuts) [reply tests/CTS/methods/requests/search/deleteRule.json:12]: I think I'll go with adding the testIndex if no testName are provided, wdyt?

**Review by millotp — COMMENTED** [review]
nice

File: templates/python/tests/client/step.mustache [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{{#isCreateClient}}
```
[the remark this answers is not in the document]
Comment (shortcuts) [reply templates/python/tests/client/step.mustache:1]: we can make the cts generator create this step file ahah

File: tests/CTS/client/common/commonApi.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
         },
         "expected": {
           "type": "userAgent",
-          "match": "^Algolia for ${{languageCased}} \\(\\d+\\.\\d+\\.\\d+(-.*)?\\)(; [a-zA-Z. ]+ (\\(\\d+((\\.\\d+)?\\.\\d+)?(-.*)?\\))?)*(; ${{clientPascalCase}} (\\(\\d+\\.\\d+\\.\\d+(-.*)?\\)))(; [a-zA-Z. ]+ (\\(\\d+((\\.\\d+)?\\.\\d+)?(-.*)?\\))?)*$"
+          "match": "^Algolia for ${{languageCased}} \\(\\d+\\.\\d+\\.\\d+(-?.*)?\\)(; [a-zA-Z. ]+ (\\(\\d+((\\.\\d+)?\\.\\d+)?(-?.*)?\\))?)*(; ${{clientPascalCase}} (\\(\\d+\\.\\d+\\.\\d+(-?.*)?\\)))(; [a-zA-Z. ]+ (\\(\\d+((\\.\\d+)?\\.\\d+)?(-?.*)?\\))?)*$"
```
[the remark this answers is not in the document]
Comment (shortcuts) [reply tests/CTS/client/common/commonApi.json:14]: ahhh yes true

File: templates/python/config.mustache [hunk 4]
```diff
@@ -34,6 +33,8 @@ class Config(BaseConfig):
         self.wait_task_time_before_retry = 100000
         {{/isSearchClient}}
 
+        UserAgent.add("{{#lambda.pascalcase}}{{client}}{{/lambda.pascalcase}}")
```
[the remark this answers is not in the document]
Comment (millotp) [reply templates/python/config.mustache:36]: nice

**Review by millotp — APPROVED** [review]
looks good !

**Pull request merged**